typescript/sdk/src/providers/transactions/transformer/ethersV5/EV5InterchainAccountTxTransformer.ts

```diff
@@ -25,29 +27,36 @@ export class EV5InterchainAccountTxTransformer
 
   public async transform(
     ...txs: PopulatedTransaction[]
-  ): Promise<PopulatedTransaction[]> {
+  ): Promise<ethers.PopulatedTransaction[]> {
     const txChainsToInnerCalls: Record<ChainName, CallData[]> = {};
 
     txs.map(({ to, data, value, chainId }: PopulatedTransaction) => {
-      assert(to, 'Invalid PopulatedTransaction: Missing to field');
-      assert(data, 'Invalid PopulatedTransaction: Missing data field');
-      assert(chainId, 'Invalid PopulatedTransaction: Missing chainId field');
       const txChain = this.multiProvider.getChainName(chainId);
       if (!txChainsToInnerCalls[txChain]) txChainsToInnerCalls[txChain] = [];
       txChainsToInnerCalls[txChain].push({ to, data, value });
     });
 
-    const transformedTxs: Promise<PopulatedTransaction>[] = [];
+    assert(
+      this.props.config.localRouter,
+      'Invalid AccountConfig: Cannot retrieve InterchainAccount.',
+    );
+
+    const interchainAccount = getInterchainAccount(
+      this.multiProvider,
+      this.props.chain,
+      this.props.config,
+    );
+
+    const transformedTxs: Promise<ethers.PopulatedTransaction>[] = [];
     Object.keys(txChainsToInnerCalls).map((txChain: ChainName) => {
-      transformedTxs.push(
-        this.props.interchainAccount.getCallRemote(
-          this.props.chain,
-          txChain,
-          txChainsToInnerCalls[txChain],
-          this.props.accountConfig,
-          this.props.hookMetadata,
-        ),
-      );
+      const transformedTx = interchainAccount.getCallRemote({
+        chain: this.props.chain,
+        destination: txChain,
+        innerCalls: txChainsToInnerCalls[txChain],
+        config: this.props.config,
+        hookMetadata: this.props.hookMetadata,
+      });
+      transformedTxs.push(transformedTx);
     });
```

Comment:
ditto here, use map
we also have an `objMap` util so you dont have to do this key value map

Reply:
ah nice– updated to use `objMap` as such:
```
    const transformedTxs: ethers.PopulatedTransaction[] = [];
    for (const txChain of objKeys(txChainsToInnerCalls)) {
      transformedTxs.push(
        await interchainAccountApp.getCallRemote({
          chain: this.props.chain,
          destination: txChain,
          innerCalls: txChainsToInnerCalls[txChain],
          config: this.props.config,
          hookMetadata: this.props.hookMetadata,
        }),
      );
    }
```

Reply:
sorry im confused
how is this using objMap?

Reply:
ah sorry about that– I read this as the objects util, not the specific method.

addressing in upcoming PR: https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3818